lib/feature/home_page/home_page.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:counter_async_redux/api/models/detailsPokemon_model.dart';
+import 'package:counter_async_redux/api/models/pokemon_model.dart';
+import 'package:counter_async_redux/api/models/specificType_model.dart';
+import 'package:flutter/material.dart';
+
+class HomePage extends StatelessWidget {
+  const HomePage({
+    required this.loadMore,
+    required this.pokemon,
+    this.next,
+  });
+
+  final VoidCallback loadMore;
+  final List<DetailsPokemon>? pokemon;
+  final String? next;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+            title: const Text("Pokédex", style: TextStyle(color: Colors.black)),
+            backgroundColor: Colors.transparent,
+            elevation: 0),
+        body: Scaffold(
+          appBar: AppBar(
+            backgroundColor: Colors.transparent,
+            elevation: 0,
+            title: Container(
+              width: double.infinity,
+              height: 40,
+              color: Colors.white,
+              child: Center(
+                child: TextField(
+                  onChanged: (query) async {},
+                  showCursor: false,
+                  decoration: const InputDecoration(
+                    border: OutlineInputBorder(
+                      borderRadius: BorderRadius.all(Radius.circular(10.0)),
+                      borderSide: BorderSide(
+                        width: 0,
+                        style: BorderStyle.none,
+                      ),
+                    ),
+                    hintText: 'Search',
+                    prefixIcon: Icon(Icons.search, color: Colors.black),
+                  ),
+                ),
+              ),
+            ),
+          ),
+          body: NotificationListener(
+            child: GridView.builder(
+              gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
+                maxCrossAxisExtent: 240,
+                childAspectRatio: 1,
+                crossAxisSpacing: 1,
+                mainAxisSpacing: 1,
+              ),
+              itemCount: pokemon?.length,
```

Comment:
assign a fallback, since pokemon is nullable, this will throw an error of pokemon is null

Reply:
@Enrickzz